wallets/react-wallet-v2/src/utils/TezosRequestHandlerUtil.ts

```diff
@@ -0,0 +1,45 @@
+import { TEZOS_SIGNING_METHODS } from '@/data/TezosData'
+import { tezosAddresses, tezosWallets } from '@/utils/TezosWalletUtil'
+import { getWalletAddressFromParams } from '@/utils/HelperUtil'
+import { formatJsonRpcError, formatJsonRpcResult } from '@json-rpc-tools/utils'
+import { SignClientTypes } from '@walletconnect/types'
+import { getSdkError } from '@walletconnect/utils'
+
+export async function approveTezosRequest(
+  requestEvent: SignClientTypes.EventArguments['session_request']
+) {
+  const { params, id } = requestEvent
+  const { request } = params
+
+  const wallet = tezosWallets[request.params.account ?? Object.keys(tezosWallets)[0]] // TODO: Select correct wallet
```

Comment:
Remaining todo here, this currently always returns the values for the first account even if the second account is requested:

<img width="400" alt="Screenshot 2023-03-09 at 14 43 38" src="https://user-images.githubusercontent.com/8663099/224043004-14264fbc-2594-4cc8-8e4a-4a6ee1435e02.png">

Reply:
Currently, the request doesn't take any arguments, meaning that we do not know which account has to be shared. I thought that the wallet would just respond with all accounts that dApp has permissions for. Does that make sense, or would it be better if we added an the address to the params of the request?

Reply:
> Currently, the request doesn't take any arguments

I'm a bit confused, for all the RPC methods other than `tezos_getAccounts` specifically the [spec](https://docs.walletconnect.com/2.0/advanced/rpc-reference/tezos-rpc#parameters-2) does have an `account` parameter and you do seem to already be passing the target account as a request parameter (e.g. [here](https://github.com/WalletConnect/web-examples/pull/127/files#diff-333319b492717cc438dc88afb2dadef1dbb71f3492dc04695a98830a8b4ac3a9R1204))

I see what you mean regarding `tezos_getAccounts`, since this kind of breaks out of the paradigm the wallet request handlers are built around (request is always directed at a specific account/signer).

In that case couldn't we simply do the following below to return all accounts?

```js
case TEZOS_SIGNING_METHODS.TEZOS_GET_ACCOUNTS:
    return formatJsonRpcResult(id, wallets.map(wallet => {
        algo: wallet.getCurve(),
        address: wallet.getAddress(),
        pubkey: wallet.getPublicKey()
    }))
```